library/Icingadb/Widget/Detail/CommentDetail.php

```diff
@@ -73,20 +73,20 @@ protected function createDetails(): array
             );
             $details[] = new HorizontalKeyValue(
                 t('Created'),
-                DateFormatter::formatDateTime($this->comment->entry_time)
+                DateFormatter::formatDateTime($this->comment->entry_time->getTimestamp())
             );
-            $details[] = new HorizontalKeyValue(t('Expires'), $this->comment->expire_time != 0
-                ? DateFormatter::formatDateTime($this->comment->expire_time)
+            $details[] = new HorizontalKeyValue(t('Expires'), $this->comment->expire_time
+                ? DateFormatter::formatDateTime($this->comment->expire_time->getTimestamp())
                 : t('Never'));
         } else {
-            if ($this->comment->expire_time != 0) {
```

Comment:
Do you have an idea why this is a non-strict comparison? Have you verified that removing it still has the same effect?

**Previously:**
NULL != 0 (false)
0 != 0 (false)
'' != 0 (false)
1 != 0 (true)

**Now:**
NULL (false)
DateTime (true)

Though, DateTime can also be `1970-01-01T00:00:00` (which is the result of `0`) and that evaluates to true now, instead of false.

You could adjust it to `!== null`, but only if `expire_time` can never be just `0`. (which would indicate that this comparison is outdated)

Reply:
Okay, verified that this cannot happen anymore. `!== null` is sufficient.